app/models/user.rb

```diff
@@ -68,6 +68,7 @@ def self.find_for_shibboleth(auth_hash, signed_in_resource=nil)
     if auth_hash.uid.blank?
       raise Avalon::MissingUserId 
     end
+    email = auth_hash.info.email || auth_hash.uid + "@ualberta.ca"
```

Comment:
It looks like the variable "email" is set, then never used ... (it likely should replace "auth_hash.info.email" on lines 75 and 76).